src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/Blobs/InstructionEncoder.cs

```diff
@@ -247,5 +248,40 @@ public void StoreArgument(int argumentIndex)
                 Builder.WriteInt32(argumentIndex);
             }
         }
+
+        public LabelHandle DefineLabel()
+        {
+            return GetBranchBuilder().AddLabel();
+        }
+
+        public void Branch(ILOpCode code, LabelHandle label)
+        {
+            // throws if code is not a branch:
+            ILOpCode shortCode = code.GetShortBranch();
+
+            GetBranchBuilder().AddBranch(Offset, label, (byte)shortCode);
+            OpCode(shortCode);
+
+            // -1 points in the middle of the branch instruction and is thus invalid.
+            // We want to produce invalid IL so that if the caller doesn't patch the branches 
+            // the branch instructions will be invalid in an obvious way.
+            Builder.WriteSByte(-1);
+        }
+
+        public void MarkLabel(LabelHandle label)
+        {
+            GetBranchBuilder().MarkLabel(Offset, label);
+        }
+
+        private BranchBuilder GetBranchBuilder()
+        {
+            if (_branchBuilderOpt == null)
+            {
+                // TODO: localize
```

Comment:
There are several more without corresponding TODO. I assume not localizing yet is to allow free flow of code between roslyn internal copy and corefx until the switchover is complete. Could you file a bug tracking the work that has to happen post-switchover?